Tweaks to serialization sections

ounsworth opened the pull request:
Closes #119 
Closes #122 
Closes #114

ounsworth committed:
Tweakes to serialization sections

ounsworth committed:
Mentioned compressed vs uncompressed encodings. Closes #122

ounsworth committed:
Tweaks

ounsworth committed:
Tweak

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -692,66 +692,55 @@ HKDF-Extract(salt="", IKM=mlkemSS || tradSS || tradCT || tradPK || Domain)
+### SerializePublicKey and DeserializePublicKey {#sec-serialize-deserialize}
 
-  bytes   The encoded public key
+Each component KEM public key is serialized according to its respective standard as shown in {{appdx_components}} and concatenated together using a fixed 4-byte length field denoting the length in bytes of the first component key, as defined below.
```
remove mention of 4-byte length field

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -692,66 +692,55 @@ HKDF-Extract(salt="", IKM=mlkemSS || tradSS || tradCT || tradPK || Domain)
-{: #alg-composite-serialize title="Composite SerializePublicKey(pk)"}
+{: #alg-composite-serialize title="SerializePublicKey(mldsaKey, tradKey) -> bytes"}
 
 Deserialization reverses this process, raising an error in the event that the input is malformed.  Each component
```
There is no error raised in the algorithm, so also change this section

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -762,65 +751,41 @@ Implicit inputs:
-
-
-## SerializePublicKey and DeserializePublicKey
+### SerializePublicKey and DeserializePublicKey
```
This whole section is a duplicate of the previous?

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -945,17 +910,17 @@ Deserialization Process:
        for all parameter sets.
 
       mldsaSeed = bytes[:64]
```
mlkemSeed

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -945,17 +910,17 @@ Deserialization Process:
   2. Output the component private keys
 
-     output (mldsaSeed, tradKey)
+     output (mldsaSeed, tradSK)
```
mlkemSeed

ounsworth replied:
This is in the deleted duplicate section, ... done now.

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -982,7 +947,7 @@ Serialization Process:
   1. Combine and output the encoded composite signature
 
-     output mlkemCT || tradEncodedSignature
+     output mlkemCT || tradCT
```
Line 944: CompositeSignatureValue -> CompositeCiphertextValue 
Line 981: signature -> ciphertext
Line 995: signature -> ciphertext
Line 991: ECDSA -> ECDH

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -895,19 +860,19 @@ Deserialization Process:
-## SerializePrivateKey and DeserializePrivateKey
+### SerializePrivateKey and DeserializePrivateKey
 
 The serialization routine for keys simply concatenates the fixed-length private keys of the component signature algorithms, as defined below:
```
signature -> KEM

janklaussner commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -982,7 +947,7 @@ Serialization Process:
 
   1. Combine and output the encoded composite signature
```
signature-> ciphertext

janklaussner reviewed (changes requested):
Whole section "SerializePublicKey and DeserializePublicKey" exists twice plus several copy&paste errors from -sigs draft

ounsworth committed:
Locked down encodings of traditional component

ounsworth committed:
Locked down encodings of traditional component

ounsworth committed:
Thanks for pointing out the copy&paste errors Jan

ounsworth commented:
Ready to merge.

janklaussner committed:
Fixes last copy&paste errors.

johngray-dev commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -699,147 +706,43 @@ HKDF-Extract(salt="", IKM=mlkemSS || tradSS || tradCT || tradPK || Domain)
+While ML-KEM has a single fixed-size representation for each of public key, private key, and ciphertext, the traditional component might allow multiple valid encodings; for example an elliptic curve public key, and therefore also ciphertext, might be validly encoded as either compressed or uncompressed [SEC1], or an RSA private key could be encoded in Chinese Remainder Theorem form [RFC8017]. In  ...
+
+* **ML-KEM**: MUST be encoded as specified in [FIPS203], using a 64-byte seed as the private key.
+* **RSA**: MUST be encoded with the `(n,e)` public key representation as specified in A.1.1 of [RFC8017] and the private key representation as specified in A.1.2 of [RFC8017]. For maximum interoperability, it is RECOMMENDED to use the `(n,d)` private key representation.
```
Same comment as in Signatures - The RSA private key format in 8017 section A.1.2 contains the fields for the CRT co-efficients and other data.  They are not optional values, but you could pass ZERO values for the integer fields if you don't want to use them.

johngray-dev replied:
Like in Sigs, we will remove the line:  For maximum interoperability, it is RECOMMENDED to use the `(n,d)` private key representation.

johngray-dev commented on draft-ietf-lamps-pq-composite-kem.md:
```diff
@@ -870,51 +774,51 @@ Implicit inputs:
   1. Parse each constituent encoded public key.
        The length of the mlkemKey is known based on the size of
-       the ML-KEM component key length specified by the Object ID
+       the ML-DSA component key length specified by the Object ID
```
This should be ML-KEM

johngray-dev committed:
Update draft-ietf-lamps-pq-composite-kem.md

Removed "For maximum interoperability, it is RECOMMENDED to use the `(n,d)` private key representation." from line 725.

janklaussner committed:
Fixes copy&paste error.

janklaussner committed:
Merge branch 'main' into mikeo_serialization
